pyccl/core.py

```diff
@@ -760,51 +704,37 @@ def _compute_linear_power_internal(self):
         # needed to init some models
         self.compute_growth()
 
-        if ((self._config_init_kwargs['transfer_function'] ==
-                'boltzmann_class') and not self.has_linear_power):
-            pk_lin = get_class_pk_lin(self)
-            psp = pk_lin.psp
-        elif ((self._config_init_kwargs['transfer_function'] ==
-                'boltzmann_isitgr') and not self.has_linear_power):
-            pk_lin = get_isitgr_pk_lin(self)
-            psp = pk_lin.psp
-        elif ((self._config_init_kwargs['transfer_function'] ==
-                'boltzmann_camb') and not self.has_linear_power):
-            pk_lin = get_camb_pk_lin(self)
-            psp = pk_lin.psp
-        else:
-            psp = None
-
-        if (psp is None and not self.has_linear_power and (
-                self._config_init_kwargs['transfer_function'] in
-                ['boltzmann_camb', 'boltzmann_class', 'boltzmann_isitgr'])):
-            raise CCLError("Either the CAMB or CLASS computation "
-                           "failed silently! CCL could not compute the "
-                           "transfer function!")
-
-        # first do the linear matter power
-        status = 0
-        status = lib.cosmology_compute_linear_power(self.cosmo, psp, status)
-        check(status, self)
-
-    def _compute_linear_power_from_arrays(self):
-        if not self._linear_power_on_input:
-            raise ValueError("Cannot compute linear power spectrum from"
-                             " input without input arrays initialized.")
-        pk_lin = Pk2D(pkfunc=None,
-                      a_arr=self.a_array_pkln,
-                      lk_arr=np.log(self.k_array_pkln),
-                      pk_arr=self.pk_array_pkln,
-                      is_logp=False,
-                      extrap_order_lok=1,
-                      extrap_order_hik=2,
-                      cosmo=None)
-
-        psp = pk_lin.psp
+        # Populate power spectrum splines
+        trf = self._config_init_kwargs['transfer_function']
+        pk = None
+        rescale_s8 = True
+        rescale_mg = True
+        if trf == 'boltzmann_class':
+            pk = get_class_pk_lin(self)
```

Comment:
IIRC you should not need to have rescale_s8 = True for any of the boltzmann code options. The python wrappers all do this already. We should double-check before removing this bit of code.

Reply:
Is that right? I thought not, based on [this comment](https://github.com/LSSTDESC/CCL/blob/6dcd05f7c3990dd7706f58d6996b1c86e3965639/pyccl/boltzmann.py#L59).

Reply:
ACK. you are right! NVM!